jar-infer/jar-infer-lib/src/main/java/com/uber/nullaway/jarinfer/BytecodeAnnotator.java

```diff
@@ -172,4 +176,55 @@ public static void annotateBytecodeInJar(
       jarOS.closeEntry();
     }
   }
+
+  /**
+   * Annotates the methods and method parameters in the classes in "classes.jar" in the given aar
+   * file with the specified annotations.
+   *
+   * @param inputZip AarFile to annotate.
+   * @param zipOS OutputStream of the output aar file.
+   * @param nonnullParams Map from methods to their nonnull params.
+   * @param nullableReturns List of methods that return nullable.
+   * @param debug flag to output debug logs.
+   * @throws IOException
+   */
+  public static void annotateBytecodeInAar(
+      ZipFile inputZip,
+      ZipOutputStream zipOS,
+      MethodParamAnnotations nonnullParams,
+      MethodReturnAnnotations nullableReturns,
+      boolean debug)
+      throws IOException {
+    BytecodeAnnotator.debug = debug;
+    LOG(debug, "DEBUG", "nullableReturns: " + nullableReturns);
+    LOG(debug, "DEBUG", "nonnullParams: " + nonnullParams);
+    for (Enumeration<? extends ZipEntry> entries = inputZip.entries();
+        entries.hasMoreElements(); ) {
+      ZipEntry zipEntry = entries.nextElement();
+      InputStream is = inputZip.getInputStream(zipEntry);
+      zipOS.putNextEntry(new ZipEntry(zipEntry.getName()));
+      if (zipEntry.getName().equals("classes.jar")) {
+        JarInputStream jarIS = new JarInputStream(is);
```

Comment:
nit: Not sure how easy it is to refactor out the code inside this if and in `annotateBytecodeInJar` to remove code duplication, but maybe worth a quick look?

Reply:
Summary of the offline discussions: 
Tried using `JarInputStream` for `annotateBytecodeInJar` case but it turns out that iterating over the entries in a Jar file using `JarInputStream` missed `MANIFEST.MF`, and this is a known [JDK bug](https://bugs.openjdk.java.net/browse/JDK-8215788). So, for now, we plan to keep this code separate and look at removing the duplication later.